crates/editor/src/items.rs

```diff
@@ -928,9 +928,18 @@ impl Item for Editor {
         &mut self,
         workspace: &mut Workspace,
         _window: &mut Window,
-        _: &mut Context<Self>,
+        cx: &mut Context<Self>,
     ) {
         self.workspace = Some((workspace.weak_handle(), workspace.database_id()));
+        cx.subscribe(
+            &workspace.weak_handle().upgrade().unwrap(),
```

Comment:
We should not use `unwrap` here, as it might panic if the workspace has been dropped. Instead, you can do something like this:

```rs  
        let weak_handle = workspace.weak_handle();
        self.workspace = Some((weak_handle, workspace.database_id()));
        if let Some(workspace) = weak_handle.upgrade() {
            cx.subscribe(&workspace, |editor, _, event: &workspace::Event, _cx| {
                if matches!(event, workspace::Event::ModalOpened) {
                    editor.mouse_context_menu.take();
                }
            })
            .detach();
        };
```